cmd/verify/verify.go

```diff
@@ -49,6 +49,14 @@ func main() {
 		log.Println("err:", err)
 	}
 
+	// Storage logici for S3 or Posix
+	var backend storage.Backend
+	if config.ArchiveType == "posix" {
+		backend = storage.NewPosixBackend(config.PosixArchive)
+	} else {
+		backend = storage.NewS3Backend(config.S3Archive, TransportConfigS3(config))
+	}
```

Comment:
This is not the way to do this.
With the logic in the proper place you will be able to do it this way:

`archive := storage.NewBacked(config.Storage)`

Which would mean that the call later on in this file would look like this:

`f := archive.ReadFile(filepath.Join(filepath.Clean(config.PosixArchive.Location), message.ArchivePath)`

Reply:
We will have this config switch on backend somewhere. I think it's good to have that as early in the runtime process as possible.

Is this what you mean all you mean? To have a function in `storage.go` like this (pseudo-go because I'm not good at go without a compiler):

```go
func NewBackend(config) Backend {
 	if config.ArchiveType == "posix" {
		return storage.NewPosixBackend(config.PosixArchive)
	} else {
		return storage.NewS3Backend(config.S3Archive, TransportConfigS3(config))
	} 
}
```